File: python/classes/classes.py
#!/usr/bin/python3

class MoneyBox:
    name = ""
    saved_amount = 0
    def save(self, amount):
        self.saved_amount += amount

piggy = MoneyBox()
piggy.name = "Miss Piggy"
print(f"Money box name: {piggy.name}")
print(f"Money box saved amount: {piggy.saved_amount}")

piggy.save(20)

print(f"Money box saved amount: {piggy.saved_amount}")

piggy.save(100)

print(f"Money box saved amount: {piggy.saved_amount}")

File: perl/classes/classes.pl
#!/usr/bin/perl

package MoneyBox;

use strict;
use warnings;

sub new {
  my ($class, $args) = @_;
  my $self = bless {
    name => "",
    saved_amount => 0
  }, $class;
}

sub save {
  my $self = shift;
  my $amount = $_[0];
  $self->{saved_amount} += $amount;
}

my $piggy = MoneyBox->new();
$piggy->{name} = "Miss Piggy";
print("Money box name: " . $piggy->{name} . "\n");
print("Money box saved amount: " . $piggy->{saved_amount} . "\n");

$piggy->save(20);

print("Money box saved amount: " . $piggy->{saved_amount} . "\n");

$piggy->save(100);

print("Money box saved amount: " . $piggy->{saved_amount} . "\n");
